docs/architecture/adr-069-unordered-account.md

```diff
@@ -0,0 +1,194 @@
+# ADR 069: Un-Ordered Nonce Lane
+
+## Changelog
+
+* Nov 24, 2023: Initial Draft
+
+## Status
+
+Proposed
+
+## Abstract
+
+Add an extra nonce "lane" to support un-ordered(concurrent) transaction inclusion.
+
+## Context
+
+Right now the nonce value (account sequence number) prevents replay-attack and make sure the transactions from the same sender are included into blocks and executed in order. But it makes it tricky to send many transactions concurrently in a reliable way. IBC relayer and crypto exchanges would be typical examples of such use cases.
+
+## Decision
+
+Add an extra nonce lane to support optional un-ordered transaction inclusion, the default lane provides ordered semantic the same as before, the new one will provide un-ordered semantic. The transaction can choose which lane to use.
+
+One of the design goals is to keep minimal overhead and breakage to the existing users who don't use the new feature.
+
+### Transaction Format
+
+It doesn't change the transaction format itself, but re-use the MSB(most significant bit) of the existing 64bits nonce value to identify the lane, `0` being the default ordered lane, `1` being the new un-ordered lane.
+
+### Account State
+
+The new nonce lane needs to add some optional fields to the account state:
+
+```golang
+type UnOrderedNonce struct {
+  Sequence uint64
+  Timestamp uint64  // the block time when the Sequence is updated
+  Gaps IntSet
+}
+
+type Account struct {
+  // default to `nil`, only initialized when the new feature is first used.
+  unorderedNonce *UnOrderedNonce
+}
+```
+
+The un-ordered nonce state includes a normal sequence value plus the set of unused(gap) values in recent history, these recorded gap values can be reused by future transactions, after used they are removed from the set and can't be used again, the gap set has a maximum capacity to limit the resource usage, when the capacity is reached, the oldest gap value is removed, which also makes the pending transaction using that value as nonce will not be accepted anymore.
```

Comment:
We use a bitmap, right? Can you explain rather, what a set bit means and what an unset bit means?

Reply:
yes, mentioned now that set bits represents the gaps.

Reply:
The gaps were not used before, so "...can be reused..." seems wrong. Maybe "...can still be used..."?